### ISSUE TITLE
list command with --depth and --json doesn't respect --depth on v3

### ISSUE THREAD

**Issue by Snugug** [npm/npm#11231]
I'm running the following command with `node 5.5.0`, `npm 3.3.12`, through `zsh 5.0.8` on `OSX 10.11.1`:

```
npm ls --depth=0
```

This works _fine_. The problem occurs when I want to output this as JSON. When I add the `--json` flag, all gets sad.

```
npm ls --depth=0 --json
```

When I run that, I get the full dependency tree not locked at `depth 0`.

However, when I use the same ZSH version on the same OS, but instead use `node 4.2.4` and `npm 2.14.12`, the JSON output is locked to `depth 0` as expected. My `package.json`'s dependencies are as follows:

```json
  "dependencies": {
    "bluebird": "^3.1.1",
    "glob": "^6.0.4"
  },
  "devDependencies": {
    "ava": "^0.10.0",
    "coveralls": "^2.11.6",
    "eslint": "^1.10.3",
    "eslint-config-xo-space": "^0.8.0",
    "nyc": "^5.3.0",
    "tap-diff": "^0.1.1"
  }
```

**Comment by othiym23** [comment]
This was fixed in [`npm@3.5.3`](https://github.com/npm/npm/releases/tag/v3.5.3). Run `npm install -g npm@3.5.3` to get it. (`npm@3.5.4` has a packaging bug that specifically affects updating to it from another version of npm, so skip that.) Closing as resolved.

**Comment by Snugug** [comment]
:+1: Thanks!
